File: android/source/VideoLocker/src/org/edx/mobile/module/registration/model/RegistrationDescription.java
package org.edx.mobile.module.registration.model;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.List;

public class RegistrationDescription {
    private @SerializedName("submit_url")     String endpoint;
    private @SerializedName("method")       String method;
    private @SerializedName("fields")       List<RegistrationFormField> fields;

    public String getEndpoint() {
        return endpoint;
    }

    public String getMethod() {
        return method;
    }

    public List<RegistrationFormField> getFields() {
        if (fields == null) {
            fields = new ArrayList<>();
        }
        return fields;
    }
}

File: OpenEdXMobile/src/main/java/org/edx/mobile/module/registration/model/RegistrationDescription.kt
package org.edx.mobile.module.registration.model

import com.google.gson.annotations.SerializedName

data class RegistrationDescription(
    @SerializedName("submit_url")
    val endpoint: String? = null,

    @SerializedName("method")
    val method: String? = null,

    @SerializedName("fields")
    var fields: List<RegistrationFormField> = listOf(),
)
